### PR TITLE
only use `install` for run_go_tool

### PR DESCRIPTION
**Pull request by kvmware**
# Changes

- :broom: Use `go install` instead of `get` for `run_go_tool` function.
- :wastebasket: remove the root check logic as `go install` does not change the `go.mod` file.

/kind cleanup

Fixes # https://github.com/knative/test-infra/issues/3113

### CODE REVIEW

File: library.sh [hunk 1]
```diff
@@ -544,26 +544,15 @@ function start_latest_eventing_sugar_controller() {
...
-      local temp_dir="$(mktemp -d)"
-      # Swallow the output as we are returning the stdout in the end.
-      pushd "${temp_dir}" > /dev/null 2>&1
-      GOFLAGS="" go ${action} "$1" || install_failed=1
-      popd > /dev/null 2>&1
-    else
-      GOFLAGS="" go ${action} "$1" || install_failed=1
-    fi
+    GOFLAGS="" go install "$1" || install_failed=1
+    echo $install_failed $1
```
Comment (chizhg) [inline library.sh:555]: Is this line needed or only for debugging?

Comment (kvmware) [reply library.sh]: Ah, good catch. removed.

**Comment by chizhg** [comment]
```
2022/04/12 23:55:54 Failed to get k8s version kubernetes version "1.21.1" is not compatible, need at least "1.22.0-0" (this can be overridden with the env var "KUBERNETES_MIN_VERSION")
```

It seems we need to update the version of `kind` CLI. Or if no one would want to create Prow jobs run with `kind` clusters, we can drop this Prow job.

**Comment by chizhg** [comment]
> /retest

You'd need to wait for a PR like https://github.com/knative/test-infra/pull/3261 to bump the image version, and then /retest.

**Comment by cardil** [comment]
I think that instead of doing the `install` and the run, we could leverage Go 1.17+ `go run package@version` construct. That will not modifying the user's PC.

File: library.sh [hunk 2]
```diff
@@ -544,26 +544,19 @@ function start_latest_eventing_sugar_controller() {
...
-    if [[ ${action} == "get" && $(pwd) == "${REPO_ROOT_DIR}" ]]; then
-      local temp_dir="$(mktemp -d)"
-      # Swallow the output as we are returning the stdout in the end.
-      pushd "${temp_dir}" > /dev/null 2>&1
-      GOFLAGS="" go ${action} "$1" || install_failed=1
-      popd > /dev/null 2>&1
-    else
-      GOFLAGS="" go ${action} "$1" || install_failed=1
-    fi
+    GOFLAGS="" go install "$package" || install_failed=1
```
Comment (dprotaso) [inline library.sh:559]: nit: for follow up - unsure if we want to drop `GOFLAGS=""`

**Comment by upodroid** [comment]
/cherrypick release-1.2

@haubenr

**Pull request merged**